spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -136,11 +157,20 @@ protected Object doReceive() {
 			String remotePath = remotePath(file);
 			Session<?> session = this.remoteFileTemplate.getSession();
 			try {
-				return getMessageBuilderFactory().withPayload(session.readRaw(remotePath))
+				AbstractIntegrationMessageBuilder<InputStream> builder = getMessageBuilderFactory()
+						.withPayload(session.readRaw(remotePath))
 						.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
 						.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
-						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
-						.build();
+						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename());
+				if (getObjectMapper() != null) {
+					try {
+						builder.setHeader(FileHeaders.REMOTE_FILE_INFO, getObjectMapper().toJson(file));
```

Comment:
Why can't we transfer `FileHeaders.REMOTE_FILE_INFO` as a regular `FileInfo` instance if we don't want JSON and everything downstream is in memory?